test/python/circuit/test_parameters.py

```diff
@@ -1178,8 +1180,32 @@ def test_compare_to_value_when_bound(self):
 
         x = Parameter("x")
         bound_expr = x.bind({x: 2.3})
+
         self.assertEqual(bound_expr, 2.3)
 
+    def test_abs_function_when_bound(self):
```

Comment:
Could you add one more test for the typical circuit workflow, something like
```python
circuit = QuantumCircuit(1)
x = Parameter(x)
circuit.p(abs(x), 0)

transpiled = transpile(circuit, basis_gates=["x", "sx", "rz"])
bound = circuit.bind_parameters({x: -1})

# check circuit is as expected
```
?

Reply:
Alternatively we could add an `abs` parameter to the `test_compile_with_ufunc` test

Reply:
Do you mean changing `test_compile_with_ufunc` test as follows?
```
    def test_compile_with_ufunc(self):
        """Test compiling of circuit with unbounded parameters
        after we apply universal functions."""
        phi1 = Parameter("phi1")
        phi2 = Parameter("phi2")
        qc = QuantumCircuit(1)
        qc.rx(numpy.abs(phi1), 0)
        qc.rx(numpy.cos(phi2), 0)
        backend = BasicAer.get_backend("qasm_simulator")
        qc_aer = transpile(qc, backend)
        self.assertIn(phi1, qc_aer.parameters)
        self.assertIn(phi2, qc_aer.parameters)
```

Reply:
I made the change.

Reply:
Hi @Cryoris, could you please review the updated tests? Thank you!

Reply:
Hi @king-p3nguin, sorry for the late reply! This doesn't look quite right, as it for example doesn't have the transpile part 🙂 The goal is to check that the `abs(parameter)` is compiled correctly. So you'd have to adjust the test below to something like
```python
theta = Parameter("theta")
qc = QuantumCircuit(1)
qc.rx(abs(theta), 0)

# transpile to different basis
transpiled = transpile(qc, basis_gates=["rz", "sx", "x", "cx"], optimization_level=0)  

bound = transpiled.bind_parameters({theta: -1})

expected = QuantumCircuit(1)
expected.rx(1.0, 0)
expected = transpile(expected, basis_gates=["rz", "sx", "x", "cx"], optimization_level=0)  

self.assertEqual(expected, bound)
```

Reply:
Thank you for the suggestion, @Cryoris ! I modified the `test_compile_with_ufunc`.